source/linux/epoll_event_loop.c

```diff
@@ -292,7 +293,19 @@ static void s_schedule_task_future(struct aws_event_loop *event_loop, struct aws
 
 static void s_cancel_task(struct aws_event_loop *event_loop, struct aws_task *task) {
     struct epoll_loop *epoll_loop = event_loop->impl_data;
-    aws_task_scheduler_cancel_task(&epoll_loop->scheduler, task);
+    if (*(volatile size_t *)&task->reserved == 1) {
```

Comment:
Why isn't this atomic? The usual pattern for this sort of thing is to compare_exchange to a flag value (which will prevent others from touching this), then lock if you are the winning writer.
Or you can read it, lock, then read again to verify it's still the same.